src/System.Diagnostics.PerformanceCounter/tests/CounterSampleCalculatorTests.cs

```diff
@@ -99,15 +99,17 @@ public static PerformanceCounter CreateCounter(string name, PerformanceCounterTy
             }
 
             DeleteCategory(name);
+            PerformanceCounterCategory.Create(category, "description", PerformanceCounterCategoryType.SingleInstance, ccdc);
```

Comment:
I realize you're experimenting at this point but I hope when you're done any sleep/retries are abstracted from the tests either by having methods like "CreateCategory" that they call that guarantee it's visible when they complete, or worst case by having a "PollUntilTrue" method they call that takes a delegate to check for the case they want and it will poll or timeout then return. Timeout should cause assert failure.